packages/next-codemod/bin/upgrade.ts

```diff
@@ -98,9 +99,20 @@ function printMigrationInstructionsAfterCodemod() {
 }
 
 export async function runUpgrade(
-  revision: string | undefined,
+  revision: string,
   options: { verbose: boolean }
 ): Promise<void> {
+  console.log(chalk.bold(`Next.js upgrade codemode (version: ${revision})`))
+
+  if (
+    !['rc', 'latest', 'canary'].includes(revision) &&
```

Comment:
I thought we were allowing versions like:

```
@next/codemod upgrade 14.0.0
```

To load from `https://registry.npmjs.org/next/14.0.0`

cc @eps1lon

Reply:
This is why I used `validateStrict` from `compare-versions`:

```tsx
import { validateStrict } from 'compare-versions'
const isValidVersion = validateStrict(version)
```

x-ref: https://github.com/vercel/next.js/pull/70599/files#diff-affeaed9f86e627e65ffc3069fc6f54711aa0eee3c197ac40603779cb96da77bR28

Reply:
I don't think we need this PR at all: https://github.com/vercel/next.js/pull/70967#pullrequestreview-2355102362